tests/aws/services/lambda_/event_source_mapping/test_lambda_integration_sqs.py

```diff
@@ -1042,6 +1043,182 @@ def test_sqs_event_source_mapping(
+                MaxNumberOfMessages=10,
+                VisibilityTimeout=120,
+                WaitTimeSeconds=5 if is_aws_cloud() else 1,
+            )
+            messages = receive_message_response.get("Messages", [])
+            for message in messages:
+                received_batch = json.loads(message["Body"])
+                batches.append(received_batch)
+                messages_to_delete.append(
+                    {"Id": message["MessageId"], "ReceiptHandle": message["ReceiptHandle"]}
+                )
+
+            aws_client.sqs.delete_message_batch(
+                QueueUrl=destination_queue_url, Entries=messages_to_delete
+            )
+            assert sum([len(batch) for batch in batches]) == 15
+            return [message for batch in batches for message in batch]
+
+        events = retry(get_msg_from_q, retries=15, sleep=5)
+        snapshot.match("Records", events)
+
+    # FIXME: this fails due to ESM not correctly collecting and sending batches
+    # where size exceeds 10 messages.
+    @markers.snapshot.skip_snapshot_verify(paths=["$..total_batches_received"])
+    @markers.aws.validated
+    def test_sqs_event_source_mapping_batching_reserved_concurrency(
+        self,
+        create_lambda_function,
+        sqs_create_queue,
+        sqs_get_queue_arn,
+        lambda_su_role,
+        snapshot,
+        cleanups,
+        aws_client,
+    ):
+        snapshot.add_transformer(snapshot.transform.sqs_api())
+        snapshot.add_transformer(SortingTransformer("Records", lambda s: s["body"]), priority=-1)
+
+        destination_queue_name = f"destination-queue-{short_uid()}"
+        function_name = f"lambda_func-{short_uid()}"
+        source_queue_name = f"source-queue-{short_uid()}"
+        mapping_uuid = None
+
+        destination_queue_url = sqs_create_queue(QueueName=destination_queue_name)
+        create_lambda_function(
+            func_name=function_name,
+            handler_file=TEST_LAMBDA_EVENT_SOURCE_MAPPING_SEND_MESSAGE,
+            runtime=Runtime.python3_12,
+            envvars={"SQS_QUEUE_URL": destination_queue_url},
+            role=lambda_su_role,
+        )
+
+        # Prevent more than 2 Lambdas from being spun up at a time
+        put_concurrency_resp = aws_client.lambda_.put_function_concurrency(
+            FunctionName=function_name, ReservedConcurrentExecutions=2
+        )
+        snapshot.match("put_concurrency_resp", put_concurrency_resp)
+
+        queue_url = sqs_create_queue(QueueName=source_queue_name)
+        queue_arn = sqs_get_queue_arn(queue_url)
+
+        create_event_source_mapping_response = aws_client.lambda_.create_event_source_mapping(
+            EventSourceArn=queue_arn,
+            FunctionName=function_name,
+            MaximumBatchingWindowInSeconds=10,
+            BatchSize=20,
+            ScalingConfig={
+                "MaximumConcurrency": 2
+            },  # Prevent more than 2 concurrent SQS workers from being spun up at a time
```

Comment:
nit: these refer to Lambda pollers I guess (also something we don't support in LS yet ;)